spec/unit/puppet/resource/catalog/certless_spec.rb

```diff
@@ -32,51 +34,43 @@
       job_id: "271036342116034393375846637943780463672"
     }
   end
-  let(:headers) do
+
+  let(:certname) { 'foo.delivery.puppetlabs.net' }
+  let(:persistence) { { facts: true, catalog: true } }
+  let(:facts) {
     {
-      'Content-Type' => 'text/json',
-      'X-Puppet-Version' => '6.4.0',
-      'Accept' => 'application/json'
+      clientcert: 'foo.delivery.puppetlabs.net',
+      clientnoop: false,
+      clientversion: '6.4.0'
     }
-  end
-  let(:response_success_body) do
+  }
+
+  let(:trusted_facts) {
     {
-      "catalog": {
-        "tags": [
-          "settings",
-          "foo.delivery.puppetlabs.net",
-          "node"
-        ],
-        "name": "foo.delivery.puppetlabs.net",
-        "version": 1554885119,
-        "catalog_uuid": "0ee685b8-0802-4a2f-94a8-077abd65880d",
-        "catalog_format": 1,
-        "environment": "production",
-        "resources": [],
-        "edges": [],
-        "classes": [
-          "settings",
-          "foo.delivery.puppetlabs.net"
-        ]
-      }
+      authenticated: 'remote',
+      certname: 'foo.delivery.puppetlabs.net',
+      domain: 'delivery.puppetlabs.net',
+      extensions: {},
+      hostname: 'foo'
     }
-  end
+  }
 
-  let(:response_error_body) do
-    "404"
-  end
-
-  describe '#headers' do
-    before do
-      allow(Puppet).to receive(:version).and_return('6.4.1')
-    end
+  let(:opts) {
+    {
+      persistence: persistence,
+      environment: environment.name,
+      facts: facts,
+      trusted_facts: trusted_facts,
+      transaction_uuid: '2078748407702309438222210184383400900',
+      job_id: '271036342116034393375846637943780463672',
+      options: {
+        prefer_requested_environment: false,
+        capture_logs: false
+      }
+    }
+  }
 
-    it { expect(indirector.headers['Content-Type']).to eq 'text/json' }
-    it { expect(indirector.headers['Accept']).to eq 'application/json' }
-    it { expect(indirector.headers['X-Puppet-Version']).to eq '6.4.1' }
-    it { expect(indirector.headers['accept-encoding']).to be_a String }
-    it { expect(indirector.headers).to include('Content-Type', 'Accept', 'X-Puppet-Version', 'accept-encoding') }
-  end
+  let(:uri) { URI.parse('https://www.example.com') }
```

Comment:
The comments below no longer really make sense https://github.com/puppetlabs/ace/pull/90/files#diff-11084da8dac421c21b37e3dbdca425308ee80df294886b83909c610ce04e2f48R77-R78